internal/api/update_credentials_v1alpha1.go

```diff
@@ -33,31 +34,28 @@ func (s *server) UpdateCredentials(
 		return nil, connect.NewError(connect.CodeUnimplemented, errSecretManagementDisabled)
 	}
 
-	credsUpdate := credentialsUpdate{
-		project:        req.Msg.GetProject(),
-		name:           req.Msg.GetName(),
-		description:    req.Msg.GetDescription(),
-		credType:       req.Msg.GetType(),
-		repoURL:        req.Msg.GetRepoUrl(),
-		repoURLISRegex: req.Msg.GetRepoUrlIsRegex(),
-		username:       req.Msg.GetUsername(),
-		password:       req.Msg.GetPassword(),
+	credType := req.Msg.GetType()
+	project := req.Msg.GetProject()
+	name := req.Msg.GetName()
+
+	if err := validateFieldNotEmpty("type", credType); err != nil {
+		return nil, err
 	}
```

Comment:
We shouldn't make `type` required here. Existing clients like the CLI only send project, name, and whatever is being updated. No value is sent for a field not receiving an update.

If this seems strange, it's only because for many of the fields, a "get" gets you a pretty heavily redacted object, so we don't expect most fields to be passed _back_ to us with the update request, because the client doesn't even know their values.